server/storage/schema/schema.go

```diff
@@ -64,3 +76,92 @@ func detectStorageVersion(lg *zap.Logger, tx backend.ReadTx) (*semver.Version, e
 	copied := V3_5
 	return &copied, nil
 }
+
+func migrateByOneVersion(lg *zap.Logger, tx backend.BatchTx, current semver.Version, upgrade bool) (*semver.Version, error) {
+	var target semver.Version
+	if upgrade {
+		target = semver.Version{Major: current.Major, Minor: current.Minor + 1}
+	} else {
+		target = semver.Version{Major: current.Major, Minor: current.Minor - 1}
+	}
+	migration := newMigration(lg, current, target)
+	err := migration.run(tx)
+	if err != nil {
+		return nil, err
+	}
+	return &target, nil
+}
+
+type migration struct {
+	current, target semver.Version
+	actions         []migrateAction
+}
+
+func newMigration(lg *zap.Logger, current semver.Version, target semver.Version) *migration {
+	var key semver.Version
+	// actions should be taken current higher version
+	if current.LessThan(target) {
+		key = target
+	} else {
+		key = current
+	}
+	actions, found := versionMigateActions[key]
+	if !found {
+		lg.Panic("version is not supported", zap.String("storage-version", target.String()))
+	}
+	return &migration{
+		current: current,
+		target:  target,
+		actions: actions,
+	}
+}
+
+func (m *migration) run(tx backend.BatchTx) error {
+	var err error
+	upgrade := m.current.LessThan(m.target)
+	tx.Lock()
+	defer tx.Unlock()
+	for _, m := range m.actions {
+		if upgrade {
+			err = m.UnsafeUpgrade(tx)
+		} else {
+			err = m.UnsafeDowngrade(tx)
+		}
+		if err != nil {
+			return err
+		}
+	}
+	// Storage version is available since v3.6, downgrading target v3.5 should clean this field.
+	if !m.target.LessThan(V3_6) {
+		UnsafeSetStorageVersion(tx, &m.target)
+	}
+	return nil
+}
+
+var versionMigateActions = map[semver.Version][]migrateAction{
+	V3_6: {
+		&addNewFieldAction{bucket: Meta, fieldName: MetaStorageVersionName, fieldValue: []byte("")},
```

Comment:
Why putting an empty "" value (instead of real value) is a proper upgrade path ? 
Probably requires a comment at list.

Reply:
This is a special case, normally it should not be empty.
I created an action to so the field will be handled by migrator (removed during downgrade).